app/src/main/java/com/projectit3048c/MainViewModel.kt

```diff
@@ -77,11 +83,56 @@ class MainViewModel(var foodService : IFoodService =  FoodService()) : ViewModel
             }
             selectedFoodAmount.foodId = document.id
             val handle = document.set(selectedFoodAmount)
-            handle.addOnSuccessListener { Log.d("Firebase", "Document Saved") }
+            handle.addOnSuccessListener {
+                Log.d("Firebase", "Document Saved")
+                if (photos.isNotEmpty()) {
+                    uploadPhotos()
+                }
+
+            }
             handle.addOnFailureListener { Log.e("Firebase", "Save failed $it ") }
         }
     }
 
+    private fun uploadPhotos() {
+        photos.forEach {
+            photo ->
+            var uri = Uri.parse(photo.localUri)
+            // nullable user entry creates technical debt. Wrap in let or enforce user on activity side.
+            val imageRef = storageReference.child("images/${user?.uid}/${uri.lastPathSegment}")
```

Comment:
Creates technical debt. Need to wrap in a let block, or enforce a user on the activity side.